src/main/java/io/pravega/connectors/hadoop/PravegaInputFormat.java

```diff
@@ -99,4 +123,43 @@ protected PravegaInputFormat(ClientFactory externalClientFactory) {
     public RecordReader<EventKey, V> createRecordReader(InputSplit inputSplit, TaskAttemptContext context) throws IOException, InterruptedException {
         return new PravegaInputRecordReader<V>();
     }
+
+    public static String fetchLatestPositionsJson(String uri, String scopeName, String streamName) throws IOException {
+        Preconditions.checkArgument(uri != null && !uri.isEmpty());
+        Preconditions.checkArgument(scopeName != null && !scopeName.isEmpty());
+        Preconditions.checkArgument(streamName != null && !streamName.isEmpty());
+
+        String pos = "";
+        try (ClientFactory clientFactory = ClientFactory.withScope(scopeName, URI.create(uri))) {
+            pos = fetchPositionsJson(clientFactory, scopeName, streamName);
+        }
+        return pos;
+    }
+
+    @VisibleForTesting
+    public static String fetchPositionsJson(ClientFactory clientFactory, String scopeName, String streamName) throws IOException {
+        BatchClient batchClient = clientFactory.createBatchClient();
+        StreamSegmentsIterator ssIter = batchClient.getSegments(new StreamImpl(scopeName, streamName), null, null);
+        StreamCut endStreamCut = ssIter.getEndStreamCut();
+
+        Gson gson = new GsonBuilder()
+            .enableComplexMapKeySerialization().create();
+        return gson.toJson(endStreamCut.asImpl().getPositions());
+    }
+
+    private StreamCut getStreamCutFromPositionsJson(String scopeName, String streamName, String s) throws IOException {
+        StreamCut sc = null;
+        if (s == null || s.isEmpty()) {
+            return sc;
```

Comment:
we could update the method signature to `Optional<StreamCut>`.

Reply:
batchClient.getSegments requires StreamCut but not Optional<StreamCut>, so better keep as it is. On the other hand, this function will be dropped after pravega/pravega#2490 is resolved.